launcher/modplatform/packwiz/Packwiz.cpp

```diff
@@ -0,0 +1,222 @@
+#include "Packwiz.h"
+
+#include <QDebug>
+#include <QDir>
+#include <QObject>
+
+#include "toml.h"
+
+#include "minecraft/mod/Mod.h"
+#include "modplatform/ModIndex.h"
+
+namespace Packwiz {
+
+// Helpers
+static inline auto indexFileName(QString const& mod_name) -> QString
+{
+    if(mod_name.endsWith(".toml"))
+        return mod_name;
+    return QString("%1.toml").arg(mod_name);
+}
+
+static ModPlatform::ProviderCapabilities ProviderCaps;
+
+auto V1::createModFormat(QDir& index_dir, ModPlatform::IndexedPack& mod_pack, ModPlatform::IndexedVersion& mod_version) -> Mod
+{
+    Mod mod;
+
+    mod.name = mod_pack.name;
+    mod.filename = mod_version.fileName;
+
+    mod.url = mod_version.downloadUrl;
+    mod.hash_format = ProviderCaps.hashType(mod_pack.provider);
+    mod.hash = "";  // FIXME
```

Comment:
yes fix you

Reply:
I can implement the Modrinth hash, since I think there's already an implementation of it readily available in Qt. But for CF, I want to leave that as a FIXME / TODO for another PR, since it's only strictly required for exporting Packwiz modpacks, which is not the aim of this PR, and I don't think there's a Murmur2 implementation on Qt we can use directly. What do you think?

Reply:
Actually I said it wrong. We can get the hashes from the mod providers, and I didn't do that, my bad! ;-;

Reply:
actually, you can get the sha1 and sha256 directly from the modrinth api. (https://docs.modrinth.com/api-spec/#operation/getVersion)

Not sure about CurseForge

Reply:
> actually, you can get the sha1 and sha256 directly from the modrinth api.

sha512* 
and yes, i've done so in https://github.com/PolyMC/PolyMC/pull/426/commits/6fe9d097f419a2920306dbabd4926ed43d4cb0d4 . Not sure if the CF one is working though, I tried using the API from their Docs, but they're using the v1 version in there, and I couldn't find an updated docs for it... Either way, as I said, CF hashing is not really all that important for this PR, and afaik we haven't decided yet what we'll do with the new CF API. If we nuke CF entirely, might as well not implement their hashes until proper packwiz support is added

Reply:
alright